pkg/handler/crud_test.go

```diff
@@ -42,6 +42,36 @@ func TestCrudFlags(t *testing.T) {
 		assert.Equal(t, "some_random_flag_key", res.(*flag.CreateFlagOK).Payload.Key)
 	})
 
+	t.Run("it should be able to create one simple flag", func(t *testing.T) {
+		res = c.CreateFlag(flag.CreateFlagParams{
+			Body: &models.CreateFlagRequest{
+				Description: util.StringPtr("simple flag"),
+				Key:         "simple_flag_key",
+				Template:    "simple",
+			},
+		})
+		assert.NotZero(t, res.(*flag.CreateFlagOK).Payload.ID)
+		assert.Equal(t, "simple_flag_key", res.(*flag.CreateFlagOK).Payload.Key)
+		flagID := uint(res.(*flag.CreateFlagOK).Payload.ID)
+		segment := entity.Segment{FlagID: flagID}
+		db.First(&segment)
+		assert.NotZero(t, segment.ID)
+		assert.Equal(t, segment.Rank, entity.SegmentDefaultRank)
+
+		variant := entity.Variant{FlagID: flagID}
+		db.First(&variant)
```

Comment:
would have preferred to do .Find() and then assert, but couldnt actually figure out how to construct the query/objects. also didnt read any kind of docs so maybe its not that hard ;)

Reply:
Yes, `First` works, since you are finding the first variant with the flagID you just created. Some syntax for query. https://gorm.io/docs/query.html